src/services/symbolDisplay.ts

```diff
@@ -164,9 +166,24 @@ namespace ts.SymbolDisplay {
 
         // Class at constructor site need to be shown as constructor apart from property,method, vars
         if (symbolKind !== ScriptElementKind.unknown || symbolFlags & SymbolFlags.Class || symbolFlags & SymbolFlags.Alias) {
-            // If it is accessor they are allowed only if location is at name of the accessor
+            // If symbol is accessor, they are allowed only if location is at declaration identifier of the accessor
             if (symbolKind === ScriptElementKind.memberGetAccessorElement || symbolKind === ScriptElementKind.memberSetAccessorElement) {
-                symbolKind = ScriptElementKind.memberVariableElement;
+                const declaration = find(symbol.declarations as ((GetAccessorDeclaration | SetAccessorDeclaration)[]), declaration => declaration.name === location);
+                if (!!declaration) {
+                    switch(declaration.kind){
+                        case SyntaxKind.GetAccessor:
+                            symbolKind = ScriptElementKind.memberGetAccessorElement;
+                            break;
+                        case SyntaxKind.SetAccessor:
+                            symbolKind = ScriptElementKind.memberSetAccessorElement;
+                            break;
+                        default:
+                            Debug.assertNever(declaration);
+                        }
+                }
+                else {
+                    symbolKind = ScriptElementKind.memberVariableElement;

```

Comment:
for my understanding: do we need to "fix" `symbolKind` again because we might have `ScriptElementKind.memberGetAccessorElement` at a setter location, because of the way `getSymbolKindOfConstructorPropertyMethodAccessorFunctionOrVar` is implemented above?

Reply:
Yes, I guess so. Do you mean I should move the logic to `getSymbolKindOfConstructorPropertyMethodAccessorFunctionOrVar`? I remember that I try to do this, but I forgot the reason why I did not do this finally.

Reply:
Aha, I remember the reason now. I thought `getSymbolKindOfConstructorPropertyMethodAccessorFunctionOrVar` tells us the kind of the symbol, and it is indeed getter/setter, then `getSymbolDisplayPartsDocumentationAndSymbolKind` decides how we display it by the location.

Reply:
But it seems there are already some logics in the function. So it is a better choice to move logic here?

Reply:
I think it's fine as it is, just wanted to make sure I understood it right. :)